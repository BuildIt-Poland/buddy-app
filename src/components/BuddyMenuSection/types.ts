import { Buddy } from 'buddy-app-schema';

export type BuddyMenuSectionProps = {
  buddy: Partial<Buddy>;
  onSelect: (id: string) => void;
};
