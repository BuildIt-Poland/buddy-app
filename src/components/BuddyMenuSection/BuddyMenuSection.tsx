import React from 'react';
import { Box, makeStyles, Typography } from '@material-ui/core';
import { Buddy } from 'buddy-app-schema';
import UserMenuListItem from 'components/UserMenuListItem';
import theme from 'styles/theme';
import { BuddyMenuSectionProps } from './types';

const useStyles = makeStyles({
  wrapper: {
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
  },
  title: {
    paddingLeft: theme.spacing(2),
  },
});

const BuddyMenuSection: React.FC<BuddyMenuSectionProps> = props => {
  const { wrapper, title } = useStyles();
  const { buddy, onSelect } = props;

  return (
    <Box className={wrapper}>
      <Typography component='h3' variant='body1' className={title}>
        <Box component='strong' fontWeight={'fontWeightBold'}>
          Buddy
        </Box>
      </Typography>
      <UserMenuListItem user={buddy as Buddy} onItemClick={onSelect} />
    </Box>
  );
};

export default BuddyMenuSection;
