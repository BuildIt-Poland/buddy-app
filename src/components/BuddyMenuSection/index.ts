import BuddyMenuSection from './BuddyMenuSection';

export default BuddyMenuSection;
