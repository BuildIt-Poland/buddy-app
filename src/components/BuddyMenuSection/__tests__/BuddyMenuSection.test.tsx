import React from 'react';
import { create } from 'react-test-renderer';

import BuddyMenuSection from '../BuddyMenuSection';

jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('../../UserMenuListItem', () => 'UserMenuListItem');

describe('Component - BuddyMenuSection', () => {
  test('renders correctly', () => {
    const component = create(
      <BuddyMenuSection
        buddy={{ id: 'id', name: 'Tom', photo: 'photo' }}
        onSelect={() => null}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
