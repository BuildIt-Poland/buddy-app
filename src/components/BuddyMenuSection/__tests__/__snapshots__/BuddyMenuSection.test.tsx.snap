// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - BuddyMenuSection renders correctly 1`] = `
<Box
  className="makeStyles-wrapper-1"
>
  <Typography
    className="makeStyles-title-2"
    component="h3"
    variant="body1"
  >
    <Box
      component="strong"
      fontWeight="fontWeightBold"
    >
      Buddy
    </Box>
  </Typography>
  <UserMenuListItem
    onItemClick={[Function]}
    user={
      Object {
        "id": "id",
        "name": "Tom",
        "photo": "photo",
      }
    }
  />
</Box>
`;
